pkg/image/importer/credentials.go

```diff
@@ -69,31 +67,50 @@ func (s *keyringCredentialStore) Basic(url *url.URL) (string, string) {
 	return basicCredentialsFromKeyring(s.DockerKeyring, url)
 }
 
-func NewCredentialsForSecrets(secrets []kapi.Secret) auth.CredentialStore {
-	return &secretCredentialStore{secrets: secrets}
+func NewCredentialsForSecrets(secrets []kapi.Secret) *SecretCredentialStore {
+	return &SecretCredentialStore{secrets: secrets}
 }
 
-type secretCredentialStore struct {
-	secrets []kapi.Secret
+func NewLazyCredentialsForSecrets(secretsFn func() ([]kapi.Secret, error)) *SecretCredentialStore {
+	return &SecretCredentialStore{secretsFn: secretsFn}
```

Comment:
How about having the underlying structure always act as lazy but on top accepting secrets (`NewCredentialsForSecrets`) or a function (`NewLazyCredentialsForSecrets`)? I'm not happy with current mixed solution.